test/test_ssm.py

```diff
@@ -134,6 +142,211 @@ def test_ssm_init_non_dirq(self):
+
+        # create a key/cert pair
+        call(['openssl', 'req', '-x509', '-nodes', '-days', '2',
+              '-newkey', 'rsa:2048', '-keyout', TEST_KEY_FILE,
+              '-out', TEST_CERT_FILE, '-subj',
+              '/C=UK/O=STFC/OU=SC/CN=Test Cert'])
+
+    @patch.object(Ssm2, '_handle_msg')
+    def test_dns_saved_to_queue(self, mock_handle_msg):
+        '''
+        Test that messages sent from different dns are sent
+        to the correct queue or dropped, depending on their dn.
+        The logging output is checked to tell us if the message
+        was sent to the correct queue, or dropped completely.
+        Therefore we don't need to create incoming or reject queues.
+        '''
+
+        # Create a list of fake valid dns that will send the messages
+        # These should be sent to the incoming queue
+        valid_dns = ("/C=UK/O=eScience/OU=CLRC/L=RAL/CN=valid-1.esc.rl.ac.uk",
+                     "/C=UK/O=eScience/OU=CLRC/L=RAL/CN=valid-2.esc.rl.ac.uk",
+                     "/C=UK/O=eScience/OU=CLRC/L=RAL/CN=valid-3.esc.rl.ac.uk")
+
+        # Create a list of fake dns that will result in a rejected message
+        # These should be sent to the rejected queue
+        rejected_dns = ("/C=UK/O=eScience/OU=CLRC/L=RAL/CN=rejected-1.esc.rl.ac.uk",
+                        "/C=UK/O=eScience/OU=CLRC/L=RAL/CN=rejected-2.esc.rl.ac.uk",
+                        "/C=UK/O=eScience/OU=CLRC/L=RAL/CN=rejected-3.esc.rl.ac.uk")
+
+        # Create a list of fake banned dns that feature in the dn list
+        # These should be dropped, and not sent to a queue
+        banned_dns = ("/C=UK/O=eScience/OU=CLRC/L=RAL/CN=banned-1.esc.rl.ac.uk",
+                      "/C=UK/O=eScience/OU=CLRC/L=RAL/CN=banned-2.esc.rl.ac.uk",
+                      "/C=UK/O=eScience/OU=CLRC/L=RAL/CN=banned-3.esc.rl.ac.uk")
+
+        # a custom empaid isn't necessary, and can just be 1
+        empaid = "1"
+
+        # Set up an openssl-style CA directory, containing the
+        # self-signed certificate as its own CA certificate, but with its
+        # name as <hash-of-subject-DN>.0.
+        p1 = Popen(['openssl', 'x509', '-subject_hash', '-noout'],
+                   stdin=PIPE, stdout=PIPE, stderr=PIPE,
+                   universal_newlines=True)
+
+        with open(TEST_CERT_FILE, 'r') as test_cert:
+            cert_string = test_cert.read()
+
+        hash_name, _unused_error = p1.communicate(cert_string)
+
+        self.ca_certpath = os.path.join(TEST_CA_DIR, hash_name.strip() + '.0')
+        with open(self.ca_certpath, 'w') as ca_cert:
+            ca_cert.write(cert_string)
+
+        # For each dn in the valid dns list,
+        # Pass it and the message to ssm and use the log output to
+        # make sure it was dealt with correctly
+        for dn in valid_dns:
+            # Capture the log output so we can use it in assertions
+            with LogCapture() as log:
+                print("Testing dn:", dn)
```

Comment:
Shouldn't generally be printing in tests. We only care about the output if something fails.

Reply:
done